apps/meteor/app/api/server/v1/moderation.ts

```diff
@@ -35,7 +37,11 @@ API.v1.addRoute(
 
 			const escapedSelector = escapeRegExp(selector);
 
-			const reports = await ModerationReports.findReportsGroupedByUser(latest, oldest, escapedSelector, { offset, count, sort }).toArray();
+			const reports = await ModerationReports.findMessageReportsGroupedByUser(latest, oldest, escapedSelector, {
+				offset,
+				count,
+				sort,
```

Comment:
Thank you for the heads up, @gabriellsh !

I understand, so we can't modify the data format returned from an endpoint unless it is a notified breaking change. That lefts three options for us:

1. As you said, modify the designs & have tabs on the main console.
2. Make two API calls, one for getting the message reports, and a second for getting the user reports, combine the results on FE, and show them on the moderation console.
3. Use a new endpoint that would return the combined data without the need for making two calls as required in the second.

I am not sure, about the product decision on changing the designs, but to me, the second solution sounds sort of hacky. What do you think would be the best approach for us?

Thanks!

Reply:
Since the moderation console design for user reports is still a wip, for now I guess we can keep the behavior as it is and make sure it's clear that current methods only deal with message reports -- and not introduce a breaking change. But this definitely will be changed depending on some design choices (I'll try to make some suggestions on it later).